src/main/java/io/tackle/pathfinder/services/AssessmentSvc.java

```diff
@@ -118,9 +114,76 @@ private Questionnaire defaultQuestionnaire() {
+        Assessment assessment = (Assessment) Assessment.findByIdOptional(assessmentId).orElseThrow(NotFoundException::new);
 
         return mapper.assessmentToAssessmentDto(assessment);
     }
 
+    @Transactional
+    public AssessmentHeaderDto updateAssessment(@NotNull Long assessmentId, @NotNull @Valid AssessmentDto assessmentDto) {
+        Assessment assessment = (Assessment) Assessment.findByIdOptional(assessmentId).orElseThrow(NotFoundException::new);
+        if (null != assessmentDto.getStakeholderGroups()) {
+            // Delete existing stakeholdergroups not included in current array
+            assessment.stakeholdergroups.forEach(stakegroup -> {
+                if (!assessmentDto.getStakeholderGroups().contains(stakegroup.stakeholdergroupId)) {
+                    log.info("Deleted stakegroup : " + stakegroup.stakeholdergroupId);
+                    stakegroup.delete();
+                }
+            });
+            // Add not existing stakeholdergroups included in the current array
+            assessmentDto.getStakeholderGroups().forEach(e -> {
+                log.info("Considering Stakeholdergroup : " + e);
+                if (assessment.stakeholdergroups.stream().noneMatch(o -> o.stakeholdergroupId == e)) {
+                    log.info("Adding Stakeholdergroup : " + e);
+                    AssessmentStakeholdergroup.builder()
+                            .assessment(assessment)
+                            .stakeholdergroupId(e)
+                            .build().persist();
+                }
+            });
+        }
+        if (null != assessmentDto.getStakeholders()) {
+            // Delete existing stakeholders not included in current array
+            assessment.stakeholders.forEach(stake -> {
+                if (!assessmentDto.getStakeholders().contains(stake.stakeholderId)) {
+                    log.info("Deleted stake : " + stake.stakeholderId);
+                    stake.delete();
+                }
+            });
+            // Add not existing stakeholders included in the current array
+            assessmentDto.getStakeholders().forEach(e -> {
+                log.info("Considering Stakeholder : " + e);
+                if (assessment.stakeholders.stream().noneMatch(o -> o.stakeholderId == e)) {
+                    log.info("Adding Stakeholder : " + e);
+                    AssessmentStakeholder.builder()
+                        .assessment(assessment)
+                        .stakeholderId(e)
+                        .build().persist();
+                }
+            });
+        }
+
+        assessmentDto.getQuestionnaire().getCategories().forEach(categ -> {
+            AssessmentCategory category = AssessmentCategory.find("assessment_questionnaire_id=?1 and id=?2", assessment.assessmentQuestionnaire.id, categ.getId()).<AssessmentCategory>firstResultOptional().orElseThrow(BadRequestException::new);
```

Comment:
check that questionnaire/category combination in Dto are attached to the correct assessment

Reply:
:+1: